src/components/Swap/Input.tsx

```diff
@@ -75,6 +75,7 @@ export default function Input({ disabled, focused }: InputProps) {
   const [inputAmount, updateInputAmount] = useSwapAmount(Field.INPUT)
   const [inputCurrency, updateInputCurrency] = useSwapCurrency(Field.INPUT)
   const inputCurrencyAmount = useSwapCurrencyAmount(Field.INPUT)
+  const [input, setInput] = useState<TokenInputHandle | null>(null)
```

Comment:
you'd think react would have a nicer way to do this

Reply:
It should be useRef but there are typing issues with the imperative handle afaict. Maybe it's fixed in v18 🤷 but the widget is v17-compatible.